lib/gc/ExecutionEngine/CPURuntime/Microkernel/BrgemmOnednn.cpp

```diff
@@ -105,25 +152,30 @@ int64_t dnnl_brgemm_dispatch(int64_t M, int64_t N, int64_t K, int64_t LDA,
   g_brgemm_desc_list.push_back(desc);
   g_brgemm_kernel_list.push_back(kernel);
   g_brgemm_palette.emplace_back(palette_buffer);
-
   return g_brgemm_desc_list.size() - 1;
 }
 
 void dnnl_brgemm_tileconfig(int64_t kernel_idx) {
-  char *palette_buffer = nullptr;
-  {
+  assert(kernel_idx >= 0 && "Invalid kernel handler");
+  auto &cache_manager = brgemm_cache_manager::getInstance();
+  brgemm_cache_info_t info;
+  if (!cache_manager.tryGet(kernel_idx, info)) {
     read_lock_guard_t g(g_brgemm_lock);
-    assert(kernel_idx >= 0 && kernel_idx < (int64_t)g_brgemm_desc_list.size() &&
+    assert(kernel_idx < (int64_t)g_brgemm_desc_list.size() &&
            "Invalid kernel handler");
-    brgemm_desc_t &desc = g_brgemm_desc_list[kernel_idx];
-    if (!desc.is_tmm) {
-      return;
-    }
-    palette_buffer = g_brgemm_palette[kernel_idx].get();
+    info = {&g_brgemm_desc_list[kernel_idx], g_brgemm_kernel_list[kernel_idx],
```

Comment:
It is not safe to assign the raw pointer to the shared_ptr in struct which will release the pointer when the `ref_count == 0`

Reply:
fixed.